File: report-generator/src/main/java/com/teamscale/report/testwise/model/TestwiseCoverageReport.java
package com.teamscale.report.testwise.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;
import java.util.List;

/** Container for coverage produced by multiple tests. */
public class TestwiseCoverageReport {

	/**
	 * If set to `true` the set of tests contained in the report don't represent the full set of tests within a
	 * partition. These tests are added or updated in Teamscale, but no tests or executable units that are missing in
	 * the report will be deleted.
	 */
	public final boolean partial;

	/** The tests contained in the report. */
	public final List<TestInfo> tests = new ArrayList<>();

	@JsonCreator
	public TestwiseCoverageReport(@JsonProperty("partial") boolean partial) {
		this.partial = partial;
	}
}

File: report-generator/src/main/kotlin/com/teamscale/report/testwise/model/TestwiseCoverageReport.kt
package com.teamscale.report.testwise.model

import com.fasterxml.jackson.annotation.JsonProperty

/** Container for coverage produced by multiple tests.  */
data class TestwiseCoverageReport(
	/**
	 * If set to `true` the set of tests contained in the report don't represent the full set of tests within a
	 * partition. These tests are added or updated in Teamscale, but no tests or executable units that are missing in
	 * the report will be deleted.
	 */
	@JvmField @param:JsonProperty("partial") val partial: Boolean
) {
	/** The tests contained in the report.  */
	@JvmField
	val tests = mutableListOf<TestInfo>()
}
